### PR TITLE
Take column into account when getting current statement query

### PR DESCRIPTION
**Pull request by MattIrv**
The LanguageService code to get a statement at a given cursor position currently doesn't take the cursor column into account. If there are multiple statements on the same line as the cursor, the current code will always return the first statement instead of finding the statement that corresponds to the cursor's column.

This PR updates that logic by adding a check to see if a statement ends before the cursor or starts after the cursor whenever the end/start line is the same line as the cursor's, and don't treat it as a match if so. It also adds a test for this situation.

### CODE REVIEW

File: test/Microsoft.SqlTools.ServiceLayer.UnitTests/QueryExecution/Execution/ServiceIntegrationTests.cs [hunk 1]
```diff
@@ -37,6 +37,24 @@ public void ExecuteDocumentStatementTest()
...
+            var workspaceService = GetDefaultWorkspaceService(query);
+            var queryService = new QueryExecutionService(null, workspaceService);
+
+            // If a line has multiple statements and the cursor is somewhere in the second one
+            var cursorColumn = statement1.Length + 1;
+            var queryParams = new ExecuteDocumentStatementParams { OwnerUri = Constants.OwnerUri, Line = 0, Column = cursorColumn };
+            var queryText = queryService.GetSqlText(queryParams);
+
+            // The text should match the second statement
+            Assert.Equal(queryText, statement2);
```
Comment (kevcunnane) [inline test/Microsoft.SqlTools.ServiceLayer.UnitTests/QueryExecution/Execution/ServiceIntegrationTests.cs]: A couple additional checks that should be quick to add:
- Verify index 0 returns statement 1
- Verify index statement1.Length (which is just to the left of the `;`) returns statement 1
- Verify the end of the text gives statement 2

I think this should be quick and cover edge cases that might come up if someone chooses to change it and accidentally add a regression in the future.

File: test/Microsoft.SqlTools.ServiceLayer.UnitTests/Profiler/ProfilerServiceTests.cs [hunk 2]
```diff
@@ -3,66 +3,68 @@
...
+// //     /// <summary>
+// //     /// Unit tests for ProfilerService
+// //     /// </summary>
+// //     public class ProfilerServiceTests
+// //     {   
+// //         /// <summary>
+// //         /// Test starting a profiling session and receiving event callback
+// //         /// </summary>
+// //         /// <returns></returns>
+// //         [Fact]
```
Comment (kburtram) [inline test/Microsoft.SqlTools.ServiceLayer.UnitTests/Profiler/ProfilerServiceTests.cs]: you can just comment out the // [Fact] line to disable the test

File: src/Microsoft.SqlTools.ServiceLayer/LanguageServices/LanguageService.cs [hunk 3]
```diff
@@ -1712,6 +1712,11 @@ internal string ParseStatementAtPosition(string sql, int line, int column)
                             if (statement.StartLocation.LineNumber <= parserLine 
                                 && statement.EndLocation.LineNumber >= parserLine)
                             {
+                                if (statement.EndLocation.LineNumber == parserLine && statement.EndLocation.ColumnNumber < parserColumn
```
Comment (kburtram) [inline src/Microsoft.SqlTools.ServiceLayer/LanguageServices/LanguageService.cs]: can also verify this works for lines such as "SELECT 1;   " with the cursor in the blank space following the ";" character?  I expect users would expect the command to work if they are on a line with a single statement, but outside what the parser strictly considers the statement.  If there are multiple statements on the line and we can determine which is the "active statement" then we should just continue.

File: test/Microsoft.SqlTools.ServiceLayer.UnitTests/Profiler/ProfilerServiceTests.cs [hunk 4]
```diff
@@ -3,66 +3,68 @@
...
+// //             requestParams.OwnerUri = testUri;
+// //             requestParams.TemplateName = "Standard";
 
-            // wait a bit for profile sessions to be polled
-            Thread.Sleep(500);
+// //             await profilerService.HandleStartProfilingRequest(requestParams, requestContext.Object);
 
-            requestContext.VerifyAll();
+// //             // wait a bit for profile sessions to be polled
+// //             Thread.Sleep(500);
```
Comment (kburtram) [inline test/Microsoft.SqlTools.ServiceLayer.UnitTests/Profiler/ProfilerServiceTests.cs]: probably just bumping this up to something like 2000 will allow time for the threads to start on the slow CI build machines..through probably we need to add test hooks to allow the product to notify the test that the polling thread has initialized.  in any case, I'm fine with disabling until we pickup the profiling work again.

File: src/Microsoft.SqlTools.ServiceLayer/LanguageServices/LanguageService.cs (hunk 3 above)
Comment (MattIrv) [reply src/Microsoft.SqlTools.ServiceLayer/LanguageServices/LanguageService.cs]: Added logic to always execute the single statement on a line if there is only one

Comment (kburtram) [reply src/Microsoft.SqlTools.ServiceLayer/LanguageServices/LanguageService.cs]: thanks!  looks good to me.
